app-client/appsync-client/src/components/UserTweets.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { graphql, compose } from 'react-apollo';
+import { propType } from 'graphql-anywhere';
+import styled from 'styled-components';
+
+import { Container } from './helpers';
+import { DeleteTweetMutation } from '../mutations';
+import { UserTweetsQuery } from '../queries';
+
+const Tweet = styled.div`
+  border-bottom: 1px solid #e6ecf0;
+  padding: 15px 15px;
+  font-size: 14px;
+  line-height: 20px;
+
+  button {
+    margin-left: 5px;
+    float: right;
+  }
+`;
+
+export class UserTweetsComponent extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.deleteTweet = this.deleteTweet.bind(this);
+  }
+
+  deleteTweet(tweet) {
+    // eslint-disable-next-line no-alert
+    if (window.confirm('Are you sure you want to delete this tweet?')) {
+      this.props.deleteTweet(tweet.tweet_id).then(() => {
+        setTimeout(() => {
+          this.props.data.refetch();
+        }, 1000); // give the backend a second to fully remove record
+      });
+    }
+  }
+
+  render() {
+    const { data } = this.props;
+    const { loading, error, getUserTwitterFeed, refetch } = data;
+
+    if (loading && !refetch) {
```

Comment:
@dankrieger first of all, thanks for helping out here!

I ran yarn and when I load the data in my case `refetch` is a Promise and therefor it will pass the condition (not rendering the loading state) and throw an error. Any ideas why it works for you?

Reply:
Oh, a cleared cache.  I think I confused refetch with the 'refetch' network status.  One moment, new commit coming.